### PR TITLE
[Docs] Add minor tweaks to migration guide 1/3

### PR DESCRIPTION
**Pull request by peter-csala**
# Pull Request

## The issue or feature being addressed

 - Part 1: Review from _Migrating execution policies_ till _Migrating timeout policies_
 - Part 2: Add **missing** _Migrating circuit breaker policies_
 - Part 3: Review from _Migrating other policies_ till _Interoperability between policies and resilience pipelines_

## Details on the issue fix or feature implementation
- _Migrating execution policies_
   - Reordered the interface list to align with code samples
   - Added line breaks to the V7's non-generic examples to have the same look and feel as the generic ones
   - Renamed the `Execute{Async}`'s `CancellationToken` parameter to `token` to avoid ambiguity 
   - Renamed lambda function's `r` parameter to `result` to align with V8's sample
   - In case of V8 used static anonymous function whenever it is possible
   - Added important section at the end as a TL;DR + link for further info
- _Migrating policy wrap_ 
  -  Added important section at the end as a TL;DR + link for further info
- _Migrating retry policies_
  - Fixed V7's code examples indentations
  - Moved and fixed the V7's `Retry forever` example
  -  In case of V8 used static anonymous function whenever it is possible
  - Added important section at the end as a TL;DR + link for further info 
- _Migrating rate limit policies_ 
   - Added important section at the end as a TL;DR + link for further info  
- _Migrating bulkhead policies_ 
   - Fixed V7's code examples indentations to align with rate limiter examples
   - Added important section at the end as a TL;DR + link for further info   
- _Migrating timeout policies_ 
   - Added important section at the end as a TL;DR + link for further info

[Preview](https://github.com/peter-csala/Polly/blob/add-minor-tweaks-to-migration-guide/docs/migration-v8.md)
  
## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/migration-v8.md [hunk 1]
```diff
@@ -27,66 +27,67 @@ This section describes how to migrate from execution policies (i.e. `IAsyncPolic
 
 In earlier versions, Polly exposed various interfaces to execute user code:
 
-- `IAsyncPolicy`
-- `IAsyncPolicy<T>`
 - `ISyncPolicy`
+- `IAsyncPolicy`
 - `ISyncPolicy<T>`
+- `IAsyncPolicy<T>`
```
Comment (martincostello) [inline docs/migration-v8.md:33]: Personally I prefer these A-Z sorted.

File: docs/migration-v8.md [hunk 2]
```diff
@@ -98,57 +99,64 @@ ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
...
+}, cancellationToken);
 ```
 <!-- endSnippet -->
 
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `ResiliencePipelineBuilder{<TResult>}` to build a resiliency pipeline
+> - Use one of the `AddXYZ` builder methods to add a new strategy to the pipeline
```
Comment (martincostello) [inline docs/migration-v8.md:150]:
```suggestion
> - Use one of the `Add*` builder methods to add a new strategy to the pipeline
```

File: docs/migration-v8.md [hunk 3]
```diff
@@ -98,57 +99,64 @@ ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
...
+> - Use either `Execute` or `ExecuteAsync` depending on the execution context
+>
+> For further information please check out the [Resilience pipelines docs](pipelines/index.md)
```
Comment (martincostello) [inline docs/migration-v8.md:153]:
```suggestion
> For further information please check out the [Resilience pipelines documentation](pipelines/index.md).
```

File: docs/migration-v8.md [hunk 4]
```diff
@@ -185,12 +193,20 @@ ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
...
-> [!NOTE]
-> See [fallback after retries](strategies/fallback.md#fallback-after-retries) for an example on how the strategies are executed.
+See [fallback after retries](strategies/fallback.md#fallback-after-retries) for an example on how the strategies are executed.
+
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `ResiliencePipelineBuilder{<TResult>}` to build a resiliency pipeline
+> - Use multiple `AddXYZ` builder methods to add new strategies to your pipeline
```
Comment (martincostello) [inline docs/migration-v8.md:203]:
```suggestion
> - Use multiple `Add*` builder methods to add new strategies to your pipeline
```

File: docs/migration-v8.md [hunk 5]
```diff
@@ -185,12 +193,20 @@ ResiliencePipeline pipeline = new ResiliencePipelineBuilder()
...
+>
+> For further information please check out the [Resilience pipelines docs](pipelines/index.md)
```
Comment (martincostello) [inline docs/migration-v8.md:205]:
```suggestion
> For further information please check out the [Resilience pipelines documentation](pipelines/index.md).
```

File: docs/migration-v8.md [hunk 6]
```diff
@@ -388,7 +404,14 @@ new ResiliencePipelineBuilder<HttpResponseMessage>().AddRetry(new RetryStrategyO
...
 
-It's important to remember that the configuration in v8 is options based, i.e. `RetryStrategyOptions` are used.
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `AddRetry` to add a retry strategy to your resiliency pipeline
+> - Use the `RetryStrategyOptions` to customize your retry behavior to meet your requirements
+>
+> For further information please check out the [Retry resilience strategy docs](strategies/retry.md)
```
Comment (martincostello) [inline docs/migration-v8.md:414]:
```suggestion
> For further information please check out the [Retry resilience strategy documentation](strategies/retry.md).
```

File: docs/migration-v8.md [hunk 7]
```diff
@@ -452,6 +475,15 @@ ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilde
 ```
 <!-- endSnippet -->
 
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `AddRateLimiter` to add a rate limiter strategy to your resiliency pipeline
+> - Use one of the derived classes of [`ReplenishingRateLimiter`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.ratelimiting.replenishingratelimiter) to customize your rate limiter behavior to meet your requirements
```
Comment (martincostello) [inline docs/migration-v8.md:483]:
```suggestion
> - Use one of the derived classes of [`ReplenishingRateLimiter`](https://learn.microsoft.com/dotnet/api/system.threading.ratelimiting.replenishingratelimiter) to customize your rate limiter behavior to meet your requirements
```

File: docs/migration-v8.md [hunk 8]
```diff
@@ -452,6 +475,15 @@ ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilde
...
+>
+> For further information please check out the [Rate limiter resilience strategy docs](strategies/rate-limiter.md)
```
Comment (martincostello) [inline docs/migration-v8.md:485]:
```suggestion
> For further information please check out the [Rate limiter resilience strategy documentation](strategies/rate-limiter.md).
```

File: docs/migration-v8.md [hunk 9]
```diff
@@ -498,6 +538,15 @@ ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilde
...
 <!-- endSnippet -->
 
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `AddConcurrencyLimiter` to add a concurrency limiter strategy to your resiliency pipeline
+> - Use the `ConcurrencyLimiterOptions` to customize your concurrency limiter behavior to meet your requirements
+>
+> For further information please check out the [Rate limiter resilience strategy docs](strategies/rate-limiter.md)
```
Comment (martincostello) [inline docs/migration-v8.md:548]:
```suggestion
> For further information please check out the [Rate limiter resilience strategy documentation](strategies/rate-limiter.md).
```

File: docs/migration-v8.md [hunk 10]
```diff
@@ -539,6 +588,15 @@ ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilde
...
 <!-- endSnippet -->
 
+> [!IMPORTANT]
+>
+> Things to remember:
+>
+> - Use `AddTimeout` to add a timeout strategy to your resiliency pipeline
+> - Use the `TimeoutStrategyOptions` to customize your timeout behavior to meet your requirements
+>
+> For further information please check out the [Timeout resilience strategy docs](strategies/timeout.md)
```
Comment (martincostello) [inline docs/migration-v8.md:598]:
```suggestion
> For further information please check out the [Timeout resilience strategy documentation](strategies/timeout.md).
```

File: docs/migration-v8.md (hunk 1 above)
Comment (martincostello) [reply docs/migration-v8.md:33]: I would say yes unless there's a good reason they're however they are now. Can you point to the other place in the docs you're referring to exactly?

Comment (peter-csala) [reply docs/migration-v8.md:33]: - [V7's Timeout](https://github.com/App-vNext/Polly/blob/main/src/Snippets/Docs/Migration.Timeout.cs)
- [V7's RateLimiter](https://github.com/App-vNext/Polly/blob/main/src/Snippets/Docs/Migration.RateLimiter.cs)
- [V7's Bulkhead](https://github.com/App-vNext/Polly/blob/main/src/Snippets/Docs/Migration.Bulkhead.cs)
- [V7's Execute](https://github.com/App-vNext/Polly/blob/main/src/Snippets/Docs/Migration.Execute.cs#L11)
- just to name a few :)

Comment (martincostello) [reply docs/migration-v8.md:33]: Ah, I thought you meant the documentation, not the code.

Comment (peter-csala) [reply docs/migration-v8.md:33]: The documentation lists these interfaces only here. Alright, I'll reorder these then.

**Review by martincostello — APPROVED** [review]

**Pull request merged**